src/Microsoft.Diagnostics.Monitoring.WebApi/DumpService.cs

```diff
@@ -36,8 +37,10 @@ public async Task<Stream> DumpAsync(IEndpointInfo endpointInfo, Models.DumpType
                 throw new ArgumentNullException(nameof(endpointInfo));
             }
 
-            // Guaranteed to not be null by StoragePostConfigureOptions.PostConfigure.
-            string dumpTempFolder = _storageOptions.CurrentValue.DumpTempFolder!;
+            StorageOptions options = _storageOptions.CurrentValue;
+            OptionsUtils.ThrowIfNotConfigured<StorageOptions>(options.Configured);
```

Comment:
This properly informs the compiler's null context because:
- `Options.ThrowIfNotConfigured` statically tell the compiler that if `options.Configured` is false it will never return.
- `options.Configured`'s getter method statically tells the compiler that if it is true then several members (e.g. `DumpTempFolder`) will not be null.

Ref: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis

Reply:
It's also important to note that I haven't found a way of generalizing this new `Configured` field behind an interface, since the interface's definition can't have the concrete-implementation's attributes for null-state static analysis.